source/dofs/dof_handler.cc

```diff
@@ -2597,9 +2586,10 @@ DoFHandler<dim, spacedim>::distribute_dofs(
     internal::DoFHandlerImplementation::Implementation::reserve_space(*this);
 
   // undo the subdomain modification
-  if (shared_tria != nullptr && shared_tria->with_artificial_cells())
-    for (const auto &cell : shared_tria->active_cell_iterators())
-      cell->set_subdomain_id(saved_subdomain_ids[cell->active_cell_index()]);
+  if (shared_tria != nullptr)
+    shared_tria->set_subdomain_mask(
+      parallel::shared::Triangulation<dim,
+                                      spacedim>::SubdomainMask::default_ids);
```

Comment:
I am not sure if it wouldn't be better to have two functions? The input argument is annoyingly long and complicated. This remembers of the the arguments passed to the constructor of the triangulation classes.

Reply:
`decltype(*shared_tria)::SubdomainMask::default_ids`